core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -915,7 +920,38 @@ struct PolicyPropertyAdaptor<WorkItemProperty::ImplWorkItemProperty<P>,
                               typename policy_in_t::traits::index_type,
                               typename policy_in_t::traits::iteration_pattern,
                               typename policy_in_t::traits::launch_bounds,
-                              WorkItemProperty::ImplWorkItemProperty<P>>;
+                              WorkItemProperty::ImplWorkItemProperty<P>,
+                              typename policy_in_t::traits::occupancy_control>;
+};
+
+template <template <class...> class Policy, class... Properties>
+struct PolicyPropertyAdaptor<DesiredOccupancy, Policy<Properties...>> {
+  using policy_in_t = Policy<Properties...>;
+  static_assert(is_execution_policy<policy_in_t>::value, "");
+  using policy_out_t = Policy<typename policy_in_t::traits::execution_space,
+                              typename policy_in_t::traits::schedule_type,
+                              typename policy_in_t::traits::work_tag,
+                              typename policy_in_t::traits::index_type,
+                              typename policy_in_t::traits::iteration_pattern,
+                              typename policy_in_t::traits::launch_bounds,
+                              typename policy_in_t::traits::work_item_property,
+                              DesiredOccupancy>;
+  static_assert(policy_out_t::experimental_contains_desired_occupancy, "");
+};
+
+template <template <class...> class Policy, class... Properties>
+struct PolicyPropertyAdaptor<MaximizeOccupancy, Policy<Properties...>> {
+  using policy_in_t = Policy<Properties...>;
+  static_assert(is_execution_policy<policy_in_t>::value, "");
+  using policy_out_t = Policy<typename policy_in_t::traits::execution_space,
+                              typename policy_in_t::traits::schedule_type,
+                              typename policy_in_t::traits::work_tag,
+                              typename policy_in_t::traits::index_type,
+                              typename policy_in_t::traits::iteration_pattern,
+                              typename policy_in_t::traits::launch_bounds,
+                              typename policy_in_t::traits::work_item_property,
+                              MaximizeOccupancy>;
```

Comment:
shouldn't there be a graph tag in here?

Reply:
ah we don't need that since graph tags are not a public one and so shouldn't exist by the time prefere/require is called right?

Reply:
https://github.com/kokkos/kokkos/pull/3379#discussion_r516292339